accepted/2021/system-drawing-win-only/system-drawing-win-only.md

```diff
@@ -0,0 +1,193 @@
+  never looked at.
+
+* There is no servicing story for bugs and security issues.
+
+However, making `libgdiplus` a first-class citizen is costly. It's around 30k
+lines of pretty old C code, virtually untested, and lacks a lot of
+functionality. `libgdiplus` also has a lot of external dependencies for all the
+image processing and text rendering, such as `cairo`, `pango`, and other native
+libraries, which makes shipping even more challenging.
+
+We've also hit some memory leaks or double free'd memory that weren't fixed in
+`libgdiplus` and that we had to work around because they were potential security
+issues. This makes it harder as we are basically on our “own” for those security
+issues. `libgdiplus` is a library we don't own, but that we must spend time
+investigating issues and proposing fixes.
+
+We have noticed that `System.Drawing.Common` is used on cross-platform mostly for
+image manipulation like QR code generators and text rendering. We haven't
+noticed heavy graphics usage as our cross-platform graphics support is very
+incomplete. For example, right now some components crash on macOS because the
+native APIs we use have been deprecated and are no longer included with recent
+macOS releases. We have never gotten a report about that crash; we discovered it
+ourselves while making `System.Drawing.Common` trim compatible.
+
+The usages we see of `System.Drawing.Common` in non-Windows environments are
+typically well supported with SkiaSharp and ImageSharp.
+
+Thus, we believe it is not worth investing in making `System.Drawing.Common` work
+on non-Windows platforms and instead propose we only keep evolving it in the
+context of Windows Forms and GDI+.
+
+## Requirements
+
+### Goals
+
+* Consumers of `System.Drawing.Common` will be informed that starting with .NET
+  6, it's only supported on Windows, via the platform compatibility analyzer
+* Cross-platform consumers can reference older versions of the package to
+  continue using `System.Drawing.Common` via `libgdiplus`
+
+### Non-Goals
+
+* Closing the gap of `System.Drawing.Common` for non-Windows platforms or
+  bundling `libgdiplus` as part of the .NET 6 installer
+
+## Stakeholders and Reviewers
+
+* Libraries team
+* ML.NET
+* Xamarin team
+
+## Design
+
+In the past we decided to include the following disclaimer in our docs:
+
+>   The `System.Drawing.Common` NuGet package works on Windows, Linux, and macOS.
+>   However, there are some platform differences. On Linux and macOS, the GDI+
+>   functionality is implemented by the `libgdiplus` library. This library is not
+>   installed by default in most Linux distributions and doesn't support all the
+>   functionality of GDI+ on Windows on macOS. There are also platforms where
+>   `libgdiplus` is not available at all. To use types from the
+>   `System.Drawing.Common` package on Linux and macOS, you must install
+>   `libgdiplus` separately. For more information, see Install .NET on Linux or
+>   Install .NET on macOS.
```

Comment:
This guidance mentions that it's not necessarily available or feature complete. Just as important is that it's not supported, which is a reliability and security concern.
That's important because it would be easy to read this and conclude that if it's available and working today for a customer, they can stick with it. We want to keep customers safe so we should explicitly recommend that they migrate off it.

Reply:
I guess you'll be updating this text anyway to explain the deprecation actions described below.

Reply:
Yes we will be updating this text and also provide extra documentation with a breaking change notice and guidance on basic samples on how to migrate to SkiaSharp or ImageSharp.